apps/frontend/src/index.tsx

```diff
@@ -12,28 +12,34 @@ import { AppRoute } from "~/libs/enums/enums.js";
 import { store } from "~/libs/modules/store/store.js";
 import { Auth } from "~/pages/auth/auth.jsx";
 
+import { ProtectedRoute } from "./libs/components/router-provider/protected-route.js";
+
 createRoot(document.querySelector("#root") as HTMLElement).render(
 	<StrictMode>
 		<ToastProvider />
 		<StoreProvider store={store.instance}>
 			<RouterProvider
 				routes={[
+					{
+						element: <Auth />,
+						path: AppRoute.SIGN_IN,
+					},
+					{
+						element: <Auth />,
+						path: AppRoute.SIGN_UP,
+					},
 					{
 						children: [
 							{
-								element: "Root",
+								element: <App />,
 								path: AppRoute.ROOT,
 							},
-							{
-								element: <Auth />,
-								path: AppRoute.SIGN_IN,
-							},
-							{
-								element: <Auth />,
-								path: AppRoute.SIGN_UP,
-							},
 						],
-						element: <App />,
+						element: (
+							<ProtectedRoute redirectTo={AppRoute.SIGN_IN}>
```

Comment:
You can omit `redirectTo` prop here as it is defaults to `AppRoute.SIGN_IN`
```suggestion
							<ProtectedRoute>
```

Reply:
I prefer we add redirectTo explicitly, in case we want to redirect to other screens at some time, not only the default one

Reply:
It probably makes sense to remove the default value if we always specify it explicitly.